pkg/compact/blocks_cleaner.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+// BlocksCleaner is a struct that deletes blocks from bucket
+// which are marked for deletion.
+type BlocksCleaner struct {
+	logger      log.Logger
+	deleteDelay time.Duration
+	bkt         objstore.Bucket
+}
+
+// NewBlocksCleaner creates a new BlocksCleaner.
+func NewBlocksCleaner(logger log.Logger, bkt objstore.Bucket, deleteDelay time.Duration) *BlocksCleaner {
+	return &BlocksCleaner{
+		logger:      logger,
+		deleteDelay: deleteDelay,
+		bkt:         bkt,
+	}
+}
+
+// DeleteMarkedBlocks reads compactor-meta.json file in block to check when the block was marked for deletion and
+// deletes the block deleteDelay duration after block is marked for deletion.
+func (s *BlocksCleaner) DeleteMarkedBlocks(ctx context.Context) error {
+	return s.bkt.Iter(ctx, "", func(name string) error {
+		id, ok := block.IsBlockDir(name)
+		if !ok {
+			return nil
+		}
+
+		deletionMark, err := metadata.ReadDeletionMark(ctx, s.bkt, s.logger, id.String())
+		if err != nil {
+			if errors.Cause(err) != metadata.ErrorDeletionMarkNotFound {
+				return errors.Wrap(err, "read file: %s")
```

Comment:
```suggestion
				return errors.Wrap(err, "read deletion mark file")
```

(`%s` doesn't do anything when using `Wrap` and not `Wrapf`, plus there is no parameter for `%s` anyway)